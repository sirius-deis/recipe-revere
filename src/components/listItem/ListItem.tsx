import { FC, PropsWithChildren } from "react";

const ListItem: FC<PropsWithChildren> = ({ children }) => {
  return <li>{children}</li>;
};

export default ListItem;
