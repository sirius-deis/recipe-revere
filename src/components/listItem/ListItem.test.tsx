import { render, screen } from "@testing-library/react";

import ListItem from "./ListItem";

describe("ListItem", () => {
  it("should match snapshot", () => {
    const { container } = render(<ListItem>Some content</ListItem>);
    expect(container).toMatchSnapshot();
  });
  it("should render plain text", () => {
    render(<ListItem>Some content</ListItem>);
    expect(screen.getByText("Some content")).toBeInTheDocument();
  });
  it("should render html", () => {
    render(
      <ListItem>
        <div>Some html</div>
      </ListItem>
    );
    expect(screen.getByText("Some html")).toBeInTheDocument();
  });
});
