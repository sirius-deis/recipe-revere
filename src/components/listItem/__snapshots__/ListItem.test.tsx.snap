// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match snapshot 1`] = `
<div>
  <li>
    Some content
  </li>
</div>
`;
